quipucords/api/aggregate_report/model.py

```diff
@@ -0,0 +1,275 @@
+            continue
+
+        if not fingerprint.cpu_core_count and source_type != DataSources.VCENTER:
+            # Do not penalize vCenter sources because they do not collect core counts.
+            aggregated.missing_cpu_core_count += 1
+
+        if not fingerprint.cpu_socket_count:
+            aggregated.missing_cpu_socket_count += 1
+
+        if not fingerprint.name:
+            aggregated.missing_name += 1
+
+        if not fingerprint.system_creation_date:
+            aggregated.missing_system_creation_date += 1
+        else:
+            system_creation_dates.append(fingerprint.system_creation_date)
+
+        cpu_core_count = fingerprint.cpu_core_count or 0
+        jboss_presence = {
+            product.name: bool(product.presence == Product.PRESENT)
+            for product in fingerprint.products.all()
+            if product.name in jboss_product_names
+        }
+        # For now (forever?) we do not care to count FUSE or BRMS.
+        # If we don't care about them, then why bother collecting the data?
+        # TODO Remove *all* other API/model/etc. code related to FUSE and BRMS?
+        jboss_eap_present = jboss_presence.get(Scan.JBOSS_EAP, False)
+        jboss_ws_present = jboss_presence.get(Scan.JBOSS_WS, False)
+        jboss_eap_cpu_core_count = cpu_core_count if jboss_eap_present else 0
+        jboss_ws_cpu_core_count = cpu_core_count if jboss_ws_present else 0
+
+        if fingerprint.infrastructure_type == SystemFingerprint.VIRTUALIZED:
+            aggregated.instances_virtual += 1
+            aggregated.jboss_eap_cores_vm += jboss_eap_cpu_core_count
+            aggregated.jboss_ws_cores_vm += jboss_ws_cpu_core_count
+        elif fingerprint.infrastructure_type == SystemFingerprint.BARE_METAL:
+            aggregated.instances_physical += 1
+            aggregated.jboss_eap_cores_physical += jboss_eap_cpu_core_count
+            aggregated.jboss_ws_cores_physical += jboss_ws_cpu_core_count
+        elif fingerprint.infrastructure_type == SystemFingerprint.HYPERVISOR:
+            aggregated.instances_hypervisor += 1
+        else:
+            aggregated.instances_unknown += 1
+        aggregated.jboss_eap_instances += 1 if jboss_eap_present else 0
+        aggregated.jboss_ws_instances += 1 if jboss_ws_present else 0
+
+        if fingerprint.cpu_socket_count:
+            # This may look strange, but I have been assured that it is correct.
+            # Some products are sold specifically with this per-two-CPU-sockets math.
+            # Try not to think too hard about this.
+            aggregated.socket_pairs += ceil(fingerprint.cpu_socket_count / 2)
+
+        if source_type == DataSources.NETWORK:
+            if not fingerprint.system_purpose:
+                aggregated.missing_system_purpose += 1
+            if not fingerprint.redhat_certs:
+                aggregated.missing_pem_files += 1
+        elif source_type == DataSources.OPENSHIFT:
```

Comment:
Jeez, the way these requirements are changing is bizarre. So now we want Openshift as part of MVP?

Reply:
(asking because the original MVP was something in the lines of "parity with the project that shall not me mentioned)